src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Android.cs

```diff
@@ -0,0 +1,473 @@
+        }
+
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2087:UnrecognizedReflectionPattern",
+            Justification = "Implementation detail of Android TimeZone")]
+        private unsafe T ReadAt<T>(long position, byte[] buffer)
+            where T : struct
+        {
+            int size = Marshal.SizeOf(typeof(T));
+            if (buffer.Length < size)
+            {
+                //TODO: Put strings in resource file
+                throw new InvalidOperationException("Internal error: buffer too small");
+            }
+
+            data!.Position = position;
+            int r;
+            if ((r = data!.Read(buffer, 0, size)) < size)
+            {
+                //TODO: Put strings in resource file
+                throw new InvalidOperationException(
+                        string.Format("Error reading '{0}': read {1} bytes, expected {2}", tzdataPath, r, size));
+            }
+
+            fixed(byte* b = buffer)
+            {
+                return (T)Marshal.PtrToStructure((IntPtr)b, typeof(T))!;
+            }
+        }
+
+        private static int NetworkToHostOrder(int value)
+        {
+            if (!BitConverter.IsLittleEndian)
+                return value;
+
+            return
+                (((value >> 24) & 0xFF) |
+                    ((value >> 08) & 0xFF00) |
+                    ((value << 08) & 0xFF0000) |
+                    ((value << 24)));
+        }
+
+        private unsafe void ReadIndex(int indexOffset, int dataOffset, byte[] buffer)
+        {
+            int indexSize = dataOffset - indexOffset;
+            int entryCount = indexSize / Marshal.SizeOf(typeof(AndroidTzDataEntry));
+            int entrySize = Marshal.SizeOf(typeof(AndroidTzDataEntry));
+
+            byteOffsets = new int[entryCount];
+            ids = new string[entryCount];
+            lengths = new int[entryCount];
+
+            for (int i = 0; i < entryCount; ++i)
+            {
+                var entry = ReadAt<AndroidTzDataEntry>(indexOffset + (entrySize*i), buffer);
+                var p = (sbyte*)entry.id;
+
+                byteOffsets![i] = NetworkToHostOrder(entry.byteOffset) + dataOffset;
+                ids![i] = new string(p, 0, GetStringLength(p, 40), Encoding.ASCII);
+                lengths![i] = NetworkToHostOrder(entry.length);
+
+                if (lengths![i] < Marshal.SizeOf(typeof(AndroidTzDataHeader)))
+                {
+                    //TODO: Put strings in resource file
+                    throw new InvalidOperationException("Length in index file < sizeof(tzhead)");
+                }
+            }
+        }
+
+        private static unsafe int GetStringLength(sbyte* s, int maxLength)
+        {
+            int len;
+            for (len = 0; len < maxLength; len++, s++)
+            {
+                if (*s == 0)
+                    break;
+            }
+            return len;
+        }
+
+        public IEnumerable<string> GetAvailableIds()
```

Comment:
I think it should just return `string[]`

Reply:
Changed, thanks!